website/source/docs/providers/consul/r/prepared_query.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: "consul"
+page_title: "Consul: consul_prepared_query"
+sidebar_current: "docs-consul-resource-prepared-query"
+description: |-
+  Allows Terraform to manage a Consul prepared query
+---
+
+# consul\_prepared\_query
+
+Allows Terraform to manage a Consul prepared query.
+
+Managing prepared queries is done using Consul's REST API. This resource is
+useful to provide a consistent and declarative way of managing prepared
+queries in your Consul cluster using Terraform.
+
+## Example Usage
+
+```
+resource "consul_prepared_query" "service-near-self" {
+    datacenter = "nyc1"
+    token = "abcd"
+    stored_token = "wxyz"
+    name = ""
+    service = "$${match(1)}"
```

Comment:
By not nesting the same as the API we'd be in trouble if we ever added a non-service PQ type on the Consul side. That's probably not super likely but worth thinking about. This is simpler to use.